service/key/key.go

```diff
@@ -263,19 +265,34 @@ func RemoteWritePasswordKey() string {
 	return "password"
 }
 
-// IsCAPICluster returns true if the cluster is in v1alpha3 and does not have the "azure-operator.giantswarm.io/version" label added by the azure operator.
-// We do not have a provider agnostic label like "giantswarm.io/version" to define this.
+// IsCAPICluster returns true if the cluster has any of the legacy labels such as azure-operator.giantswarm.io/version.
 func IsCAPICluster(obj metav1.Object) bool {
 	// TODO once we have migrated all clusters to CAPI, we can remove this
-	switch v := obj.(type) {
-	case *capiv1alpha3.Cluster:
-		if _, ok := v.Labels["azure-operator.giantswarm.io/version"]; ok {
+
+	checker := func(labels map[string]string) bool {
+		if _, ok := labels["azure-operator.giantswarm.io/version"]; ok {
+			return false
+		}
+		if _, ok := labels["release.giantswarm.io/version"]; ok {
```

Comment:
Maybe I'm missing something, but we are definitely adding the `release.giantswarm.io/version` label to CAPI clusters. For example, on AWS https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/templates/_helpers.tpl#L29

Reply:
ok this is my mistake. I couldn't test with CAPI so I had to go yolo

Reply:
https://github.com/giantswarm/prometheus-meta-operator/pull/885